src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -0,0 +1,28 @@
+import type { QueryRawOptions, Sequelize } from '../../sequelize';
+import type { AbstractQueryGenerator } from './query-generator';
+
+export interface QueryInterfaceOptions {
+  sequelize: Sequelize;
+  queryGenerator: AbstractQueryGenerator;
+}
+
+export class AbstractQueryInterfaceTypeScript {
+  readonly sequelize: Sequelize;
+  readonly queryGenerator: AbstractQueryGenerator;
+
+  constructor(options: QueryInterfaceOptions) {
+    this.sequelize = options.sequelize;
+    this.queryGenerator = options.queryGenerator;
+  }
+
+  /**
+   * Queries the schema (table list).
+   *
+   * @param schema The schema to query. Applies only to Postgres.
+   * @param options
+   */
+  async createSchema(schema: string, options?: QueryRawOptions): Promise<void> {
```

Comment:
I think we do need to extend these with `CreateSchemaQueryOptions`, right? Or are they already included in `QueryRawOptions`?